code/modules/mob/living/simple_animal/abnormality/waw/greed_king.dm

```diff
@@ -0,0 +1,178 @@
+	melee_damage_upper = 80
+	can_breach = TRUE
+	threat_level = WAW_LEVEL
+	start_qliphoth = 1
+	work_chances = list(
+						ABNORMALITY_WORK_INSTINCT = list(25, 25, 50, 50, 55),
+						ABNORMALITY_WORK_INSIGHT = 0,
+						ABNORMALITY_WORK_ATTACHMENT = list(0, 0, 50, 50, 55),
+						ABNORMALITY_WORK_REPRESSION = list(0, 0, 40, 40, 40)
+						)
+	work_damage_amount = 10
+	work_damage_type = RED_DAMAGE
+	//Some Variables cannibalized from helper
+	var/charge_check_time = 1 SECONDS
+	var/charge_check_cooldown
+	var/charging = FALSE
+	var/dash_num = 50	//Mostly a safeguard
+	var/list/been_hit = list()
+	var/busy = FALSE
+
+
+	can_spawn = FALSE // For testing only
+
+	ego_list = list(
+		/datum/ego_datum/weapon/goldrush,
+		/datum/ego_datum/armor/goldrush
+		)
+	gift_type =  /datum/ego_gifts/goldrush
+
+
+
+/mob/living/simple_animal/hostile/abnormality/greed_king/AttackingTarget()
+	if(charging)
+		return
+
+/mob/living/simple_animal/hostile/abnormality/greed_king/Move()
+	return FALSE
+
+/mob/living/simple_animal/hostile/abnormality/greed_king/breach_effect(mob/living/carbon/human/user)
+	..()
+	icon = 'ModularTegustation/Teguicons/64x48.dmi'
+	//Center it on a hallway
+	pixel_y = -8
+	base_pixel_y = -8
+
+	teleport()	//Let's Spaghettioodle out of here
+
+/mob/living/simple_animal/hostile/abnormality/greed_king/proc/teleport()
+	if(charging || busy)
+		return
+
+	//set busy, animate and teleport
+	busy = TRUE
+	var/turf/T = pick(GLOB.xeno_spawn)
+	animate(src, alpha = 0, time = 5)
+	SLEEP_CHECK_DEATH(5)
+	animate(src, alpha = 255, time = 5)
+	forceMove(T)
+
+	//Clear lists
+	been_hit = list()
+	charge_check()
+	if(!charging)
+		SLEEP_CHECK_DEATH(5 SECONDS)
+		busy = FALSE
+		teleport()
+
+/mob/living/simple_animal/hostile/abnormality/greed_king/proc/charge_check()
+	//targeting
+	var/mob/living/carbon/human/target
+	if(charging && !busy)
+		return
+	busy = TRUE
+	var/list/possible_targets = list()
+	for(var/mob/living/carbon/human/H in view(20, src))
+		possible_targets += H
+	if(!LAZYLEN(possible_targets))
+		return
+	if(!charging && !target)
+		target = pick(possible_targets)
+	//Start charge
+	SLEEP_CHECK_DEATH(2 SECONDS)
+	if(ishuman(target)&& !charging)
+		var/dir_to_target = get_cardinal_dir(get_turf(src), get_turf(target))
+		if(dir_to_target in list(NORTH, SOUTH, WEST, EAST))
+			charge(dir_to_target, 0, target)
+			return
+
+/mob/living/simple_animal/hostile/abnormality/greed_king/proc/charge(move_dir, times_ran, target)
+	setDir(move_dir)
+	var/stop_charge = FALSE
+	if(times_ran >= dash_num)
+		stop_charge = TRUE
+	var/turf/T = get_step(get_turf(src), move_dir)
+	if(!T)
+		charging = FALSE
+		return
+	if(T.density)
+		stop_charge = TRUE
+	for(var/obj/structure/window/W in T.contents)
+		W.obj_destruction()
+	for(var/obj/machinery/door/D in T.contents)
+		if(D.density)
+			stop_charge = TRUE
+	for(var/mob/living/simple_animal/hostile/abnormality/D in T.contents)	//This caused issues earlier
```

Comment:
Does it suddenly stop charging if it detects an abnormality? You may need to specify that the abnormality is not herself. I had this issue with Snow white where her attack would effect herself too. 
I might be wrong but she might be going so fast that she detects herself infront of herself and stops. Which then would make her stop charge and teleport.

Reply:
If im right then if(D.density && !istype(D, /mob/living/simple_animal/hostile/abnormality/greed_king)) Might help? I might be wrong.